### PR TITLE
SOLR-14024 Invalid html generated by changes2html.pl

### PR DESCRIPTION
**Pull request by asalamon74**
Original lucene-solr pull request: apache/lucene-solr#1060

# Description

changes2html.pl generates invalid html

# Solution

Fix it to generate a valid html.

# Tests

Created the html and tested it using https://validator.w3.org/

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/CHANGES.txt [hunk 1]
```diff
@@ -10028,7 +10028,6 @@ Upgrading from Solr 5.1
     SOLR_SSL_TRUST_STORE_PASSWORD, SOLR_SSL_NEED_CLIENT_AUTH and SOLR_SSL_WANT_CLIENT_AUTH
     have been introduced. The bin/solr scripts configure the SOLR_SSL_OPTS property
     automatically based on the above new properties.
-
```
Comment (janhoy) [inline solr/CHANGES.txt:10031]: Unrelated change

**Review by janhoy — COMMENTED** [review]
Manual review looks good.

Please indicate a quick way to test the change in practice, i.e. command to invoke the script and what to look for, i.e. can we run a tool on the resulting HTML file to see how the former style was broken and the new is ok?

Is there any visual change in the output, or just "correctness"?

**Comment by asalamon74** [comment]
I'm calling the following to invoke the script:

```
./gradlew changesToHtml
```
It generates the html which can be found in `./solr/documentation/build/site/changes/Changes.html`.

I was using https://validator.w3.org/#validate_by_input as a validator a copied the generated html to it. For the original HTML it gave me a couple hundred errors like:

- Self-closing syntax (/>) used on a non-void HTML element. Ignoring the slash and treating as a start tag.
- Element ul not allowed as child of element ul in this context. (Suppressing further errors from this subtree.)
- No li element in scope but a li end tag seen.

The solr/CHANGES.txt change is not unrelated. That empty line caused a problem in the generation (the script was starting a new block because of that).

Visually I tried to keep it as close as possible, there are minor changes, like marginsize.

**Comment by janhoy** [comment]
Ok, tried it and it looks good. The one (three) errors still there is due to the huge license text comment which consumes >1024 chars, not sure if there is anything to do with that - actually I think we are allowed to use a smaller license text for files like this, see https://www.apache.org/legal/src-headers.html#is-a-short-form-of-the-source-header-available

> "Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements; and to You under the Apache License, Version 2.0. "

Although I think this is not a valid case to use it. No big deal.

**Comment by asalamon74** [comment]
Interesting, because I don't remember running into this license problem, but @rmuir mentioned a very similar thing in the lucene part of this change: https://github.com/apache/lucene/pull/31

**Comment by janhoy** [comment]
Thanks András!
